WordPress/gutenberg
# Calendar Block: Add color supports and polish styles

Fix: #40989

## What?
This PR adds color block support to the calendar block and updates the style to match a variety of themes.

## Why?
This block has hard-coded text color and border color.
This causes problems that the text is difficult to read and the design doesn't match the other blocks, depending on the background color of the content.

## How?
I've made the following changes based on a table block (using the same `table` tag):

- Remove hard-coded colors
- Border color inherit text color
- Make `thead` bottom border thicker
- Add text and link color with block support

## Testing Instructions

- Check out this branch, and build sources.
- Insert a calendar block in the site editor ( or the post editor ).
- Change the global style and see how the appearance of the calendar block changes.
- If possible, check with classic themes or other block themes as well.

## Screenshots or screencast 

https://user-images.githubusercontent.com/54422211/176364901-def33894-5464-4493-ace7-2565912b27ff.mp4
